testsuite/tests/contrib/lkt_semantic/array_type/test.lkt

```diff
@@ -0,0 +1,45 @@
+# TODO: support @invalid
```

Comment:
I guess I can remove this comment and uncomment all the `@invalid` lines below? Since it prints diagnostics we should be able to tackle regressions if any.

Reply:
> Thanks for your work, Laurent!
> 
> Whenever it is possible, we try to put tests for a feature/bugfix in the same commit as the implementation: can you merge the two commits?
> 
> Also, can you add the TN to the commit message, for tracking purposes? Thanks!

You're welcome ;). Sure I'll do that!